haskell/test/Test/Ligo/BaseDAO/Common.hs

```diff
@@ -156,30 +157,30 @@ originateLigoDaoWithConfig
  -> OriginateFn cep m
 originateLigoDaoWithConfig extra config qt = do
 
-  owner1 :: Address <- refillable $ newAddress "owner1"
-  operator1 :: Address <- refillable $ newAddress "operator1"
-  owner2 :: Address <- refillable $ newAddress "owner2"
-  operator2 :: Address <- refillable $ newAddress "operator2"
+  owner1 <- refillable $ newAddress "owner1"
+  operator1 <- refillable $ newAddress "operator1"
+  owner2 <- refillable $ newAddress "owner2"
+  operator2 <- refillable $ newAddress "operator2"
 
-  admin :: Address <- refillable $ newAddress "admin"
+  admin <- refillable $ newAddress "admin"
```

Comment:
Might as well create these in bulk:

```haskell
owner1 ::< operator1 ::< owner2 ::< operator2 ::< admin ::< Nil' <- refillables $
  newAddresses $ "owner1" ::< "operator1" ::< "owner2" ::< "operator2" ::< "admin" ::< Nil'
```

Reply:
For some reason the compilare figures this to be a failable pattern and refuses to accept for want of `MonadFail` instance for the `ClevelandBaseMonad`.

Reply:
Ugh. Yeah, GHC isn't smart enough to figure out this pattern isn't actually fallible in this context. We do provide `MonadFail` in scenarios, so adding `MonadFail m` will work, but apparently you'd need to add it in a lot of places elsewhere that use this function, which is quite annoying. You could get away with using a pattern-match in a lambda, although it's somewhat ugly:

```haskell
  refillables (newAddresses $ "owner1" :< "operator1" :< "owner2" :< "operator2" :< "admin" :< Nil)
  >>= \(owner1 :< operator1 :< owner2 :< operator2 :< admin :< Nil) -> do
    owner1Balance <- getBalance owner1
-- ...
```